src/Service.Tests/SqlTests/GraphQLFilterTests/GraphQLFilterTestBase.cs

```diff
@@ -71,6 +71,100 @@ public async Task TestStringFiltersEqWithMappings(string dbQuery)
             SqlTestHelper.PerformTestEqualJsonStrings(expected, actual.ToString());
         }
 
+        /// <summary>
+        /// Tests correct rows are returned with filters containing 2 varchar columns one with null and one with non-null values. 
+        /// </summary>
+        [TestMethod]
+        public async Task TestFilterForVarcharColumnWithNullAndNonNullValues()
+        {
+            string graphQLQueryName = "journals";
+            string gqlQuery = @"{
+                journals( " + QueryBuilder.FILTER_FIELD_NAME + @" : { and: [ {color: {isNull: true}} {ownername: {eq: ""Abhishek""}}]})
+                {
+                    items {
+                        id
+                        journalname
+                        color
+                        ownername
+                    }
+                }
+            }";
+
+            string dbQuery = MakeQueryOn(
+                "journals",
```

Comment:
How are we ensuring the perf issue identified doesnt regress again?

Reply:
since this perf issue was happening only for very large set of records ~20 million. we can't add a test for this atm.

Reply:
can't we test that the parameters are being created with the exact/expected type and that we're not losing information by using `DbType`?